### ISSUE TITLE
Check for unsupported use of `synchronized` in async blocks

### ISSUE THREAD

**Issue by retronym** [scala/scala-async#152]
I think we should treat it as a by-name method for the purposes of our analysis / ANF transform.

```
scala> ObjectTpe.decl(TermName("synchronized")).defString
res2: String = final def synchronized[T0](x$1: T0): T0
```

**Comment by SethTisue** [comment]
@retronym should this be closed now that scala/scala#9082 has been merged?

**Comment by retronym** [comment]
Yep, thanks
